src/ol/source/ogcTileUtil.js

```diff
@@ -399,7 +399,15 @@ function parseTileMatrixSet(
       }
     }
 
-    Object.assign(localContext, context);
+    Object.assign(
+      localContext,
+      {
+        z: localContext.tileMatrix,
+        x: localContext.tileCol,
+        y: localContext.tileRow,
+      },
+      context,
+    );
```

Comment:
Note: This only take care of situation where API use z/x/y but in theory the spec is more open that this. That's said I don't see a better solution for now